scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java

```diff
@@ -433,28 +426,89 @@ public JobId submit(Job userJob)
+                }
+            });
+            return blockUntilJobIdInitialized(userJob, job, submitting);
         } catch (Exception e) {
             logger.warn("Error when submitting job.", e);
             throw e;
         }
     }
 
+    /**
+     * This method blocks until we got job id, or exception occurred.
+     *
+     * if future is done but it got an exception -> throw exception
+     * if future is done but no exception occurred
+     * if id is not initialized throw exception, it means we have some severe bug
+     * if id is initialized it is fine we return id
+     * if future is not done
+     * if id is initialized -> register as waiting and return id
+     * if id is not initialize we continue sleep
+     */
+    private JobId blockUntilJobIdInitialized(Job userJob, InternalJob job, Future<?> submitting) throws JobCreationException {
+        try {
+            while (true) {
+                if (submitting.isDone()) {
+                    try {
+                        submitting.get();
+                    } catch (ExecutionException execException) {
+                        // if future is done but it got an exception -> throw exception
+                        throw new JobCreationException(execException);
+                    }
+                    // if future is done but no exception occurred
+                    if (job.getId().longValue() == InternalJob.UNINITIALIZED_JOB_ID) {
+                        String errorMessage = String.format("Submission of job %s is finished, but job id is not properly initialized", userJob.getName());
+                        logger.error(errorMessage);
+                        // if id is not initialized throw exception, it means we have some severe bug
+                        throw new JobCreationException(errorMessage);
+                    } else {
+                        // if id is initialized it is fine we return id
+                        return job.getId();
+                    }
+                } else { // if future is not done
+                    // if id is initialized -> register as waiting and return id
+                    if (job.getId().longValue() != InternalJob.UNINITIALIZED_JOB_ID) {
+                        registerJobAsUnderSumbission(job.getId());
+                        return job.getId();
+                    } else {
+                        // if id is not initialize we continue sleep
+                        Thread.sleep(10);
+                    }
+                }
+            }
+        } catch (InterruptedException e) {
+            logger.warn("SchedulerFrontend was interrupted", e);
+            throw new JobCreationException(e);
+        }
+    }
+
+
     /**
      * {@inheritDoc}
      */
     @Override
     public List<String> getUserSpaceURIs() throws NotConnectedException, PermissionException {
         UserIdentificationImpl ident = frontendState.checkPermission("getUserSpaceURIs",
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)